bolt-design-system/bolt
# Modal | JS | Add page scrolling option

## Jira

http://vjira2:8080/browse/BDS-1681

## Summary

Allow page scroll behind modal to prevent content from shifting.

## Details

Until we have a solid solution for content shifting / page scrolling when modal is open, just let the page scroll. The previous work to prevent the page from scrolling is now conditional based on the `preventBodyScrolling` prop.

Code changes:
- Hide the modal "custom events" page that demos how to set padding on fixed elements when modal opens/closes. Will bring this back later if necessary.
- Move scrollbar functions into Core (per feedback in #1296).
- Move scrollbar calculation functions to `BoltModal` class, so that they are only called once  (per feedback in #1296).
- Add `preventBodyScroll` as prop, default to `false`.


## How to test

- Run locally in Chrome, Safari, and IE.
- Verify that there is no page content shift when modal opens.
- Verify that the page scrolls behind the modal (when it is long enough).
- Flag any issues.
